init-tools.sh

```diff
@@ -15,6 +15,43 @@ __PROJECT_JSON_FILE=$__PROJECT_JSON_PATH/project.json
 __PROJECT_JSON_CONTENTS="{ \"dependencies\": { \"Microsoft.DotNet.BuildTools\": \"$__BUILD_TOOLS_PACKAGE_VERSION\" }, \"frameworks\": { \"netcoreapp1.0\": { } } }"
 __INIT_TOOLS_DONE_MARKER=$__PROJECT_JSON_PATH/done
 
+# Modified platform detection logic from dotnet/cli/scripts/obtain/dotnet-install.sh 16692fc
```

Comment:
FYI @markwilkie @maririos we still need to figure out a solution for all this platform detection code we have to put in many scripts across many repos.